bayes_opt/util.py

```diff
@@ -44,38 +44,49 @@ def acq_max(ac, gp, y_max, bounds, random_state, constraint=None, n_warmup=10000
     :return: x_max, The arg max of the acquisition function.
     """
 
+    # We need to adjust the acquisition function to deal with constraints when there is some
+    if constraint is not None:
+
+        def adjusted_ac(x, gp, y_max):
+            """Acquisition function adjusted to fulfill the constraint when necessary"""
+            values = ac(x, gp=gp, y_max=y_max)
+            p_constraints = constraint.predict(x)
+
+            # Slower fallback for the case where any values are negative
+            if np.any(values < 0):
+                # TODO: This is not exactly how Gardner et al do it.
+                # Their way would require the result of the acquisition function
+                # to be strictly positive, which is not the case here. For a
+                # positive target value, we use Gardner's version. If the target
+                # is negative, we instead slightly rescale the target depending
+                # on the probability estimate to fulfill the constraint.
+                return np.array(
+                    [
+                        value * p if value > 0 else value / (0.5 + 0.5 * p)
+                        for value, p in zip(values, p_constraints)
+                    ]
+                )
```

Comment:
We could vectorize this as well (much much faster) if we can ensure that always all values have the same sign (positive or negative). Is the sign a property of the acquisition function?